oemer/build_system.py

```diff
@@ -66,29 +72,29 @@ class Key(enum.Enum):
 
 
 class Voice:
-    def __init__(self):
-        self.id: int = None
+    def __init__(self) -> None:
+        self.id: int | Any = None
         self.note_ids: list[int] = []
-        self.stem_up: bool = None
-        self.group_id: int = None
-        self.x_center: float = None
-        self.label: NoteType = None
-        self.has_dot: bool = None
-        self.group: int = None
-        self.track: int = None
-        self.duration: int = None
-        self.rhythm_name: str = None
-
-    def init(self):
+        self.stem_up: bool | Any = None
+        self.group_id: int | Any = None
+        self.x_center: float | Any = None
+        self.label: NoteType | Any = None
+        self.has_dot: bool | Any = None
+        self.group: int | Any = None
+        self.track: int | Any = None
+        self.duration: int | Any = None
+        self.rhythm_name: str | Any = None
```

Comment:
I'm concern about the grammar here that `int | Any` is only available for python3.10 or above. To be compatible with lower version, I would suggest rewrite in this way
```
from typing import Union

class Voice:
  def init(self):
    self.stem_up: Union[bool, None] = None
    ...
```

Reply:
Please also fix other places as well.

Reply:
Done. This was actually one of the biggest headaches. To share that: The proper way to deal with the none initialization would be to either add a lot more none checks in the code or make sure that the objects get properly initialized and thus avoiding none entirely. Both approaches appeared too risky to me for the moment, and therefore I decided to work with union types and type ignores.